rmw_connext_cpp/src/functions.cpp

```diff
@@ -164,62 +212,73 @@ rmw_create_publisher(
     rmw_set_error_string("callbacks handle is null");
     return NULL;
   }
-  std::string type_name = std::string(callbacks->package_name) + "::msg::dds_::" +
-    callbacks->message_name + "_";
+  std::string type_name = _create_type_name(callbacks, "msg");
+  // Past this point, a failure results in unrolling code in the goto fail block.
```

Comment:
Yes it is straight forward, but very hard to quickly introspect. Using the same target label means that you can't search/jump quickly without checking that you're in the same scope. There's 12 instances in this PR. 

xyz_cleanup where xyz are unique would seem to be a better pattern so that you can quickly do a search and audit that that specific set of goto's without having to mentally filter the other scopes.